### PR TITLE
Add support for RFC6020 for "if-feature", "when" and "must" statements

### PR DESCRIPTION
**Pull request by gcampbell512**
The "if-feature statement is defined in RFC6020 section 7.18.2 The "when" statement is defined in RFC6020 section 7.19.5 The "must" statement is defined in RFC6020 section 7.5.3

Added a number of tests to test the new functionality.

### CODE REVIEW

**Comment by gcampbell512** [comment]
This patch requires another patch in apteryx-xml before it can be used.

File: data.c [hunk 1]
```diff
@@ -210,6 +248,23 @@ _sch_gnode_to_xml (sch_instance * instance, sch_node * schema, sch_ns *ns, xmlNo
         return NULL;
     }
 
+    flags |= SCH_F_YANG_CONDITIONS;
+    sch_check_yang_condition (schema, node, flags, &path, &condition);
+    if (condition)
+    {
+    printf ("%s:%u path %s condition %s\n", __func__, __LINE__, path, condition);
```
Comment (tony-vanderpeet) [inline data.c]: clean up before delivery

File: data.c [hunk 2]
```diff
@@ -210,6 +248,23 @@ _sch_gnode_to_xml (sch_instance * instance, sch_node * schema, sch_ns *ns, xmlNo
...
+        if (!sch_process_yang_condition (netconf_get_g_schema (), node, path, condition))
+        {
+            printf ("%s:%u path %s condition %s\n", __func__, __LINE__, path, condition);
```
Comment (tony-vanderpeet) [inline data.c]: clean up before delivery

**Review by tony-vanderpeet — DISMISSED** [review]
Couple of stray printfs to remove - otherwise happy that this change is applying the new code in apteryx-xml in the right places.

**Review by tony-vanderpeet — DISMISSED** [review]

**Review by tony-vanderpeet — APPROVED** [review]

**Pull request merged**